### ISSUE TITLE
Update dependencies being generated in the package.json

### ISSUE THREAD

**Issue by ramya-rao-a** [Azure/autorest.typescript#600]
https://github.com/Azure/azure-sdk-for-js/pull/7883 is an attempt to onboard a package generated by this code generator to use rush. We hit the below roadblocks
- Older tslib, typescript, rollup
- The need to specify a version for `@types/node` due to the below error when running `tsc`

```
../../../common/temp/node_modules/.pnpm/registry.npmjs.org/@types/tunnel/0.0.0/node_modules/@types/tunnel/index.d.ts:6:23 - error TS4090: Conflicting definitions for 'node' found at 'C:/code/sdk-for-js/common/temp/node_modules/.pnpm/registry.npmjs.org/@types/node/13.9.0/node_modules/@types/node/ts3.2/index.d.ts' and 'C:/code/sdk-for-js/common/temp/node_modules/.pnpm/registry.npmjs.org/@types/node/8.10.59/node_modules/@types/node/ts3.2/index.d.ts'. Consider installing a specific version of this library to resolve the conflict.
```

The question to the code generator team is 
- How often do we keep updating the deps and release new versions of the code generator?
- Can we have `@types/node` added as a dev dep in the generated code

cc @KarishmaGhiya, @chradek, @joheredi

**Comment by daviwil** [comment]
Honestly this makes me wonder if the code generator should actually live in the azure-sdk-for-js repo so that we can keep all of the core dependencies up to date.  If part of the code generation logic is to look at the code generator's package.json to pick out the versions for things then maybe that'd help everything stay up to date naturally.  Thoughts?

**Comment by chradek** [comment]
I think I agree with @daviwil that it'd be easiest to keep dependencies in sync if the generator were in this repo, though I think we should only consider moving the v6 generator (I think v5 and lower uses git submodules).

We can update autorest.typescript as often as we need to as far as I know, but I'm worried about how we'll find out about dependency changes that need to make it back into the generator if they are in 2 separate repos. Unless we had a way of triggering a PR on the autorest.typescript repo whenever the dependencies of some sort of template package got updated... 🤔

**Comment by daviwil** [comment]
Yep definitely just v6.  Funny you mentioned cross-repo triggering, that's what happens when you open a PR to `ms-rest-js`, it actually runs the tests for `autorest.typescript` to ensure that changes to the pipeline didn't cause bugs in the generated code.  It'd be nice to do something similar in our repo and that'd be a lot easier if the code was there.

However, now that I say that I realize that this isn't a common pattern for the AutoRest generators.  Even in the new v3 world everyone still has theirs in separate repos.  We'd probably have to talk to the overall AutoRest group to see how people felt about that (or if it matters for languages to be consistent on that front).
